ytypes/node.go

```diff
@@ -51,10 +51,14 @@ type retrieveNodeArgs struct {
 	// If val is set to a non-nil value, leaf/leaflist node corresponding
 	// to the given path is updated with this value.
 	val interface{}
+	// useJSONEncoding indicates that the input value is in JSON encoding
+	// instead of gNMI's typed encoding.
+	useJSONEncoding bool
```

Comment:
General question -- should we support this mode like this, or rather using the `TypedValue` field that corresponds to JSON?

Reply:
Thanks for catching this, I agree that we should just drop this mode and just add support for unmarshalling `TypedValue`s that contain json_ietf. This fits the current use case better as well.

The use case I'm looking at is that in the current Ondatra implementation, `Marshal7951` is used to marshal messages instead of `EncodeTypedValue`, which means leaves are also in JSON format. Do you think Ondatra should be using `EncodeTypedValue` instead?

Reply:
Since this implementation is very different from just supporting `json_ietf`, I will drop this and open another PR.